cli/src/command.rs

```diff
@@ -268,13 +268,27 @@ pub fn run() -> Result<()> {
 		Some(Subcommand::Benchmark(cmd)) => {
 			let runner = cli.create_runner(cmd)?;
 			let chain_spec = &runner.config().chain_spec;
-
 			set_default_ss58_version(chain_spec);
-
-			Ok(runner.sync_run(|config| {
-				cmd.run::<service::kusama_runtime::Block, service::KusamaExecutor>(config)
-				.map_err(|e| Error::SubstrateCli(e))
-			})?)
+			let spec_name = runner.config().impl_name.to_lowercase();
+
+			if spec_name.contains("polkadot") {
+				Ok(runner.sync_run(|config| {
+					cmd.run::<service::polkadot_runtime::Block, service::PolkadotExecutor>(config)
+						.map_err(|e| Error::SubstrateCli(e))
+				})?)
+			} else if spec_name.contains("kusama") {
+				Ok(runner.sync_run(|config| {
+					cmd.run::<service::kusama_runtime::Block, service::KusamaExecutor>(config)
+						.map_err(|e| Error::SubstrateCli(e))
+				})?)
+			} else if spec_name.contains("westend") {
```

Comment:
maybe we can fix that common problem by only allowing benchmarks with `*-dev`

Reply:
Given your explanation, I'll probabkly do that. So we never ever ever want to run benchmarks with any chain other than `*-dev`?

Reply:
probably not.